src/test/db_benchmark/multi_file_impl.ml

```diff
@@ -0,0 +1,52 @@
+open Core
+
+module Make () : Common.Database = struct
+  type t = { root : string }
+
+  let name = "multi_file"
+
+  let create directory = Unix.mkdir_p directory ; { root = directory }
+
+  let close _t = ()
+
+  (* Get block number from key ID *)
+  let block_of_key key = key / Common.keys_per_block
+
+  (* Get offset within block from key ID *)
+  let offset_of_key key = key mod Common.keys_per_block
+
+  (* Get file path for a block *)
+  let block_path t block_num =
+    Filename.concat t.root (Printf.sprintf "%d.block" block_num)
+
+  (* Calculate byte offset within file for a key *)
+  let byte_offset_of_key key = offset_of_key key * Common.value_size
+
+  let set_block t ~block_num values =
+    let path = block_path t block_num in
+    (* Write all values directly to file without in-memory concatenation *)
+    Out_channel.with_file path ~binary:true ~f:(fun oc ->
+        List.iter values ~f:(Out_channel.output_string oc) )
```

Comment:
This is not fair competition. Files are written to a same handler in a same channel without close/open. 

But say with LMDB, we're creating a txn per write.